contracts/erc20-gateway/ERC20Cogateway.sol

```diff
@@ -82,4 +83,35 @@ contract ERC20Cogateway is MasterCopyNonUpgradable, GenesisERC20Cogateway, Messa
 
         activated = true;
     }
+
+    /**
+     * @notice It verifies that ERC20Gateway contract exists on origin chain
+     *         using merkle account proof.
+     *
+     * @param _blockNumber Block number at which ERC20Gateway contract is to
+     *                     be proven.
+     * @param _rlpAccount RLP encoded account node object.
+     * @param _rlpParentNodes RLP encoded value of account proof node array.
+     *
+     * \post Emits `GatewayProven` event with `messageInbox` and
+     *       `_blockNumber` parameters.
+     * \post Calls `MessageInbox.proveStorageAccount()` function with
```

Comment:
Ordering of post actions should be reversed. 
First proveStorageAccount is called and then event is emitted.

Reply:
Done